v3.0_roadmap.adoc

```diff
@@ -0,0 +1,243 @@
+= Road map for James 3.0
+
+This document aims at dividing existing and upcoming code into the following categories :
+
+ - supported
+ - experimental
+ - unsupported
+
+== Mailbox section
+
+.Feature matrix : mailbox backend
+|===
+|Feature name |Status | Comment
+
+|JPA
+|supported
+|Already existing code
+
+|Memory
+|Unsupported
+|Reference for testing
+
+|Cassandra
+|experimental
+|Essentially we do not know about how performance will behave. Also consistency issues might arise.
+
+|ElasticSearch
+|supported
+|
+
+|HBase
+|unsupported
+|
+
+|JCR
+|unsupported
+|
+
+|Lucene
+|unsupported
+|
+
+|Caching
+|unsupported
+|
+
+|Maildir
+|unsupported
+|Known issue on MailboxPath handling
+
+|===
+
+What being "supported implies for a backend ?
+
+ - Being MPT tested
+ - Deployement tests part of unit tests (ex : cassandra-guice)
+ - Having full support in server/data/data-@backend@
+ - Having its mappers passing generic unit tests for mappers
+ - It should at least support USER_FLAGS and NAMESPACE features on MPT tests
+ - Be able to handle a constant gatling workload for at least 4 hours with constant latency
+ - Be tested with thunderbird (we should write scenari for that)
+
+
+.Feature matrix : mailbox capabilities
+|===
+|Feature name |Status | Comment
+
+|ACL
+|unsupported
+|
+
+|Shared mailbox
+|unsupported
+|
+
+|Quota
+|experimental
+|
+
+|User flags
+|supported
+|
+
+|Distributed events
+|Unsupported
+|
+
+|===
+
+=== Short note about what needs to be DONE
+
+**General** :
+
+ - Define Thunderbird scenari
+ - Write Gatling tests for IMAP
+
+**Cassandra** :
+
+ - We need a Cassandra implementation for MailRepository
+
+**Memory** :
+
+ - We need a Memory SieveRepository
+ - We need a Memory implementation for MailRepository
+
+**JPA** :
+
+ - We need JAMES-1693-2 to be merged.
+ - We need JPA implementation for SieveRepository and MailRepository.
+ - We need JPA to valid MessageMapper and MailboxMapper unit tests.
+
+Bonus : have quota support with JPA...
+
+== Server
+
+.Injection framework
+|===
+|Name |Status
+
+|Spring
+|Supported
+
+|Guice cassandra
```

Comment:
I would like a supported Guice JPA. We should try hard to get contributions on the parts of our codebase we like  the most.